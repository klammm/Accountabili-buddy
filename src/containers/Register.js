import React, { Component } from 'react';
import { View, TextInput, StyleSheet, Text, TouchableOpacity } from 'react-native';

export default class Register extends Component {
  submit = (values) => {
    console.log('submitting form', values)
  }

  render() {
    return (
      <View style={styles.container}>
        <TextInput style={styles.input} placeholder="First Name"/>
        <TextInput style={styles.input} placeholder="Last Name"/>
        <TextInput style={styles.input} placeholder="Username"/>
        <TextInput style={styles.input} placeholder="Password"/>
        <TextInput style={styles.input} placeholder="Confirm Password"/>
        <TouchableOpacity style={styles.button}>
          <Text style={styles.buttonText}>Submit</Text>
        </TouchableOpacity>
      </View>
    )
  }
}

Register.navigationOptions = {
  title: "Register"
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#F5FCFF',
  },
  button: {
    backgroundColor: '#0091EA',
    borderWidth: 2,
    borderRadius: 10,
    borderColor: '#0091EA',
    padding: 5
  },
  buttonText: {
    backgroundColor: '#0091EA',
    color: "white",
    height: 20,
    textAlign: 'center',
    width: 150
  },
  input: {
    borderColor: 'black',
    borderWidth: 1,
    height: 35,
    margin: 5,
    textAlign: 'center'
  }
})
