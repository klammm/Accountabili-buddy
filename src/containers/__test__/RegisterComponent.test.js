import 'react-native';
import React from 'react';
import { shallow } from 'react-test-renderer';
import toJson from 'enzyme-to-json';
import { Register } from '../Register';

import renderer from 'react-test-renderer';

describe('tests the register component', () => {

  it('renders Register component', () => {
    const component = renderer.create(
      <Register
        firstNameChanged={jest.fn}
        lastNameChanged={jest.fn}
        usernameChanged={jest.fn}
        registerEmailChanged={jest.fn}
        registerPasswordChanged={jest.fn}
        registerUser={jest.fn}
        confirmPasswordChanged={jest.fn}
        passwordMatch={jest.fn}
        loginUser={jest.fn}
        firstName={'firstName'}
        lastName={'lastName'}
        username={'username'}
        registerEmail={'registerEmail'}
        registerPassword={'registerPassword'}
        confirmedPassword={'confirmedPassword'}
        matchedPassword={'true'}
        navigation= {'Login'}
      />
    );
    expect(toJson(component)).toMatchSnapshot();
  });
});
