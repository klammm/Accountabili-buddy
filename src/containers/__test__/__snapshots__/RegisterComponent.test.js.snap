// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests the register component renders Register component 1`] = `undefined`;
